src/api/transports/http.js

```diff
@@ -42,12 +43,58 @@ export function jsonRpc(uri, {method, id, params}) {
 export default class HttpTransport extends Transport {
   send(api, data, callback) {
     if (this.options.useAppbaseApi) {
-        api = 'condenser_api';
+      api = 'condenser_api';
     }
     debug('Steem::send', api, data);
     const id = data.id || this.id++;
     const params = [api, data.method, data.params];
-    jsonRpc(this.options.uri, {method: 'call', id, params})
-      .then(res => { callback(null, res) }, err => { callback(err) })
+    const retriable = this.retriable(api, data);
+    const fetchMethod = this.options.fetchMethod;
+    if (retriable) {
+      retriable.attempt((currentAttempt) => {
+        jsonRpc(this.options.uri, { method: 'call', id, params, fetchMethod }).then(
+          res => { callback(null, res); },
+          err => {
+            console.error('An error occurred hitting the Steem API:', err);
+            if (retriable.retry(err)) {
+              console.errror('Retrying...');
+              return;
+            }
+            callback(retriable.mainError());
+          }
+        );
+      });
+    } else {
+      jsonRpc(this.options.uri, { method: 'call', id, params, fetchMethod }).then(
+        res => { callback(null, res); },
+        err => { callback(err); }
+      );
+    }
+  }
+
+  get nonRetriableOperations() {
+    return this.options.nonRetriableOperations || [
+      'broadcast_transaction',
+      'broadcast_transaction_with_callback',
+      'broadcast_transaction_synchronous',
+      'broadcast_block',
+    ];
+  }
+
+  // An object which can be used to track retries.
+  retriable(api, data) {
+    if (this.nonRetriableOperations.some((o) => o === data.method)) {
+      // Do not retry if the operation is non-retriable.
+      return null;
+    } else if (this.options.retry) {
+      // If `this.options.retry` is a map of options, use it. If
+      // `this.options.retry` is `true`, use default options.
+      return (Object(this.options.retry) === this.options.retry) ?
```

Comment:
The comments here seemed a little out of order; if I understand it correctly, is this an equivalent replacement?

```
    if (this.nonRetriableOperations.some((o) => o === data.method)) {
      // Do not retry if the operation is non-retriable.
      return null;
    } else if (Object(this.options.retry) === this.options.retry) {
      // If `this.options.retry` is a map of options, use it.
      return retry.operation(this.options.retry);
    } else if (this.options.retry) {
      // If `this.options.retry` is `true`, use default options.
      return retry.operation();
    } else {
      // Otherwise, don't retry.
      return null;
    }
```